### PR TITLE
Fix/Order dialog state

### LINKED ISSUE #837: Popup response for submission failed and successful is very similar
Steps to reproduce:
- Put limit order IOC for price under the current market price so that it does not get hit
- Approved in wallet with enough assets to fulfill order
- At first glance, even though the submission failed, which is also in the text, the popup screen is very similar with colour and (including check V) with an order that is successful 
I have included two images, one that is successful and one that failed.

Screenshot 2022-07-21 at 12 37 59
Screenshot 2022-07-21 at 12 30 16

### PR DESCRIPTION
**Pull request by mattrussell36**
# Related issues 🔗

Closes #837 

# Description ℹ️

- Fixes a problem where failed orders were being rendered in a success state in the order dialog
- Refactors order hooks and dialogs to be more re-usable. Decoupling the order dialog from the generic vega transaction dialog
- Refactors to return vega transaction dialog from the hook so we dont need to rely on useEffects to control state

### CODE REVIEW

File: libs/ui-toolkit/src/components/dialog/dialog.tsx [hunk 1]
```diff
@@ -50,15 +52,20 @@ export function Dialog({
...
-              {title}
-            </h1>
-          )}
-          {children}
+          <div className="flex gap-12 max-w-full">
+            {icon && <div className="pt-8 fill-current">{icon}</div>}
+            <div data-testid="dialog-content" className="flex-1">
+              {title && (
+                <h1
+                  className={`text-h4 font-bold text-black-95 dark:text-white-95 mt-0 mb-6 capitalize ${titleClassNames}`}
```
Comment (MadalinaRaicu) [inline libs/ui-toolkit/src/components/dialog/dialog.tsx:60]:
```suggestion
                  className={`text-h4 font-bold text-black-95 dark:text-white-95 mt-0 mb-6  ${titleClassNames}`}
```

Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/dialog/dialog.tsx:60]: all dialogs have now capitalized letters, and Pete will ask me to change that as it is not consistent with the rest of the console style 

Screenshot 2022-07-23 at 10 13 04


Screenshot 2022-07-23 at 10 12 56

File: libs/wallet/src/vega-transaction-dialog/vega-transaction-dialog.tsx [hunk 2]
```diff
@@ -159,108 +98,77 @@ export const VegaDialog = ({
...
+
+  return null;
 };
 
-interface OrderDialogWrapperProps {
-  children: ReactNode;
-  icon: ReactNode;
-  title: string;
-}
+const getIntent = (transaction: VegaTxState) => {
```
Comment (MadalinaRaicu) [inline libs/wallet/src/vega-transaction-dialog/vega-transaction-dialog.tsx:131]: (blocking): pulled the branch and tested. there is still sth not working correctly yet, the order has failed and we have a rejection reason but still displaying successful intent of the dialog

Screenshot 2022-07-23 at 10 05 40

File: libs/orders/src/lib/order-hooks/use-order-cancel.tsx [hunk 3]
```diff
@@ -1,116 +1,76 @@
-import { useCallback, useEffect, useRef, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
```
Comment (MadalinaRaicu) [inline libs/orders/src/lib/order-hooks/use-order-cancel.tsx:1]:
```suggestion
import { useCallback, useState } from 'react';
```

Comment (MadalinaRaicu) [reply libs/orders/src/lib/order-hooks/use-order-cancel.tsx:1]: useEffect not used

File: libs/wallet/src/vega-transaction-dialog/vega-transaction-dialog.tsx [hunk 4]
```diff
@@ -1,155 +1,94 @@
...
   }
 
-  // Transaction error
   if (transaction.status === VegaTxStatus.Error) {
     return (
-      <OrderDialogWrapper
-        title="Order rejected by wallet"
-        icon={<Icon name="warning-sign" size={20} />}
-      >
+      <div data-testid={transaction.status}>
```
Comment (MadalinaRaicu) [inline libs/wallet/src/vega-transaction-dialog/vega-transaction-dialog.tsx:75]: ok, so I have tried to follow the [designs]( https://www.figma.com/file/K49Nx7XfocShhHcwwQzQkm/Console-V2?node-id=1652%3A26053) in Figma. we need to be aware we are changing these as we can see. 



image


Transaction Complete


VEGA TRADING PORTFOLIO

File: libs/deal-ticket/src/components/deal-ticket-manager.tsx [hunk 5]
```diff
@@ -16,27 +19,18 @@ export const DealTicketManager = ({
...
-        return 'Order filled';
-      case OrderStatus.PartiallyFilled:
-        return 'Order partially filled';
-      case OrderStatus.Parked:
-        return 'Order parked';
-      default:
-        return 'Submission failed';
-    }
-  };
+  const { submit, transaction, finalizedOrder, reset, Dialog } =
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/deal-ticket-manager.tsx:22]:
```suggestion
  const { submit, transaction, finalizedOrder, Dialog } =
```

**Review by MadalinaRaicu — CHANGES_REQUESTED** [review]

**Review by edd — APPROVED** [review]
👍 Reviewed together in person at the same time as #851. This encapsulation is neat and I like it.

**Review by MadalinaRaicu — APPROVED** [review]

**Pull request merged**